File: codeforces/82A/cpp/82a.cpp
#include <iostream>
#include <string>
using namespace std;

const int c_num = 5;
const string c_names[c_num] = {"Sheldon", "Leonard", "Penny", "Rajesh", "Howard"};

int main()
{
    int cur = c_num, x = 1;
    int n;
    cin >> n;
    n--;
    while (n >= cur) {
        n -= cur;
        x *= 2;
        cur = c_num * x;
    }
    cout << c_names[n / x] << endl;

    return 0;
}

File: codeforces/82A/go/82a.go
package main

import "fmt"


const c_num = 5
var c_names = []string{"Sheldon", "Leonard", "Penny", "Rajesh", "Howard"}

func main() {
    var n int
    fmt.Scan(&n)

    cur := c_num
    x := 1

    n -= 1
    for n >= cur {
        n -= cur
        x *= 2
        cur = c_num * x
    }

    fmt.Println(c_names[n / x])
}
